packages/library-legacy/src/connection.ts

```diff
@@ -3188,6 +3196,458 @@ export class Connection {
+  private castToSupply(value: any): Supply {
+    return {
+      total: Number(value.total),
+      circulating: Number(value.circulating),
+      nonCirculating: Number(value.nonCirculating),
+      nonCirculatingAccounts: value.nonCirculatingAccounts.map(
+        (a: string) => new PublicKey(a),
+      ),
+    };
+  }
+
+  /**
+   * EXPERIMENTAL
+   * Cast the experimental `GetAccountInfoApiResponseWithDefaultData`
+   * response to the legacy `AccountInfo<Buffer>`
+   *
+   * The type `GetAccountInfoApiResponseWithDefaultData` is not exported
+   */
+  private castToAccountInfoBuffer(
+    value: any | null,
+  ): AccountInfo<Buffer> | null {
+    if (value) {
+      const data = value.data
+        ? Buffer.from(bs58.decode(value.data))
+        : Buffer.alloc(0);
+      return (
+        value && {
+          executable: value.executable,
+          owner: new PublicKey(value.owner),
+          lamports: Number(value.lamports),
+          data,
+          rentEpoch: Number(value.rentEpoch),
+        }
+      );
+    }
+    return null;
+  }
+
+  /**
+   * EXPERIMENTAL
+   * Cast the experimental `GetVoteAccountsApiResponse<TVoteSupply>`
+   * response to the legacy `VoteAccountStatus`
+   *
+   * The type `GetVoteAccountsApiResponse<TVoteSupply>` is not exported
+   */
+  private castToVoteAccountStatus(value: any): VoteAccountStatus {
+    return {
+      current: value.current.map((a: any) => {
+        return {
+          ...a,
+          activatedStake: Number(a.activatedStake),
+          lastVote: Number(a.lastVote),
+        };
+      }),
+      delinquent: value.delinquent.map((a: any) => {
+        return {
+          ...a,
+          activatedStake: Number(a.activatedStake),
+          lastVote: Number(a.lastVote),
+        };
+      }),
+    };
+  }
+
+  /**
+   * EXPERIMENTAL
+   * Cast the experimental `GetInflationRewardApiResponse`
+   * response to the legacy `InflationReward`
+   *
+   * The type `GetInflationRewardApiResponse` is not exported
+   */
+  private castToInflationReward(value: any): (InflationReward | null)[] {
+    return value.map((a: any) => {
+      return (
+        a && {
+          epoch: Number(a.epoch),
+          effectiveSlot: Number(a.effectiveSlot),
+          amount: Number(a.amount),
+          postBalance: Number(a.postBalance),
+          commission: Number(a.commission),
+        }
+      );
+    });
+  }
+
+  /**
+   * EXPERIMENTAL
+   * Cast the experimental `GetEpochInfoApiResponse`
+   * response to the legacy `EpochInfo`
+   *
+   * The type `GetEpochInfoApiResponse` is not exported
+   */
+  private castToEpochInfo(value: any): EpochInfo {
+    return {
+      epoch: Number(value.epoch),
+      slotIndex: Number(value.slotIndex),
+      slotsInEpoch: Number(value.slotsInEpoch),
+      absoluteSlot: Number(value.absoluteSlot),
+      blockHeight: Number(value.blockHeight),
+      transactionCount: value.transactionCount
+        ? Number(value.transactionCount)
+        : undefined,
```

Comment:
You're doing this to cast from bigint to `Number`, right?

I planned for this!

What you have to do is to create an `IRPCTransport` that does _not_ have the bigint conversion feature.

Check this out: https://github.com/solana-labs/solana-web3.js/blob/9968a730655a79ec1bb25b665eb40a9ce4300b6a/packages/library/src/rpc-transport.ts#L21

In the _legacy_ library, just create a transport _without_ the request coalescer. At that point, a _lot_ more of these types should just be usable as-is.

Reply:
smhhhhhhh thank you lol. I had to read a lot of how you built `library` before I realized exactly how it works

Reply:
Sorry; I linked you to the wrong line earlier. Updated the comment above!